### PR TITLE
Create jwt

### PR DESCRIPTION
**Pull request by SimonLab**
ref: #3
Create a jwt with the email of the user added as a claim.

### CODE REVIEW

File: config/prod.secret.exs [hunk 1]
```diff
@@ -27,6 +27,7 @@ config :auth_mvp, AuthMvpWeb.Endpoint,
   http: [:inet6, port: String.to_integer(System.get_env("PORT") || "4000")],
   secret_key_base: secret_key_base
 
+config :joken, default_signer: System.get_env("SECRET_KEY_BASE")
```
Comment (nelsonic) [inline config/prod.secret.exs:30]: Would be good to add `SECRET_KEY_BASE` environment variable to `.env_sample` file so people know they need it. 👍

File: lib/auth_mvp_web/controllers/github_auth_controller.ex [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
   use AuthMvpWeb, :controller
-
   @elixir_auth_github Application.get_env(:auth_mvp, :elixir_auth_github) || ElixirAuthGithub
   @doc """
   `index/2` handles the callback from GitHub Auth API redirect.
   """
   def index(conn, %{"code" => code, "state" => client}) do
-    {:ok, _profile} = @elixir_auth_github.github_auth(code)
+    {:ok, profile} = @elixir_auth_github.github_auth(code)
+    jwt = AuthMvp.Token.generate_and_sign!(%{email: profile.email})
```
Comment (nelsonic) [inline lib/auth_mvp_web/controllers/github_auth_controller.ex:9]: Was `AuthMvp.Token.generate_and_sign` previous defined? (_it does not appear in this PR_)

Comment (SimonLab) [reply lib/auth_mvp_web/controllers/github_auth_controller.ex:9]: The jwt functions are imported in `AuthMvp.Token` module via the `use` of the `Joken.Config` module:
https://github.com/dwyl/auth-mvp/pull/4/files#diff-aeabf9b66d4c4f2b10a9da4bd106fb78

It is the recommended way to import these functions from Joken: https://hexdocs.pm/joken/configuration.html#module-approach
image

File: config/prod.secret.exs (hunk 1 above)
Comment (SimonLab) [reply config/prod.secret.exs:30]: Added on the PR #10

File: lib/auth_mvp_web/controllers/github_auth_controller.ex (hunk 2 above)
Comment (nelsonic) [reply lib/auth_mvp_web/controllers/github_auth_controller.ex:9]: Macro. 👍

File: config/prod.secret.exs (hunk 1 above)
Comment (nelsonic) [reply config/prod.secret.exs:30]: Thanks. 👍

**Review by nelsonic — APPROVED** [review]
@SimonLab looks good. 👍

**Pull request merged**